Final/multinblkserver.cpp

```diff
@@ -0,0 +1,129 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <netdb.h>
+#include "ThreadPool.h"
+#include "fdlist.h"
+
+const int PORT_NO = 8080;
+const int BACKLOG_SIZE = 2;
+
+// Prototypes
+void select(int sockfd, fd_set *read_fds_ref, struct node *head);
+bool wasEchoed(fd_set *read_fds_ref, struct node *head);
+void echo(int fd);
+int accept(int sockfd);
+
+void error(const char *msg)
+{
+    perror(msg);
+    exit(0);
+}
+
+int main(int argc, char const *argv[])
+{
+	// Step 1: Set up socket
+    int sockfd = socket(AF_INET, SOCK_STREAM, 0); // Create socket file descriptor for server
+    if (sockfd < 0) 
+        error("ERROR opening socket");
+
+	// Step 2: Bind
+	struct sockaddr_in host;
+	host.sin_family = AF_INET; // IPv4 connection
+	host.sin_port = htons(PORT_NO); // bound to port 8080
+	host.sin_addr.s_addr = INADDR_ANY; // connected to localhost
+	const struct sockaddr *host_ptr = (struct sockaddr*) &host;
+	int bound = bind(sockfd, host_ptr, sizeof(host)); // bind sockfd to port
+	if (bound < 0)
+		error("ERROR binding to socket");
+
+	// Step 3: Listen
+	int listening = listen(sockfd, BACKLOG_SIZE); // The socket will actively listen for incoming connections and queue them up
+	if (listening < 0)
+		error("ERROR listener failed");
+
+	fd_set read_fds;
+	struct node *head = createFDList(0); // Create empty fdlist (file descriptor list)
+	while (1) {
+		select(sockfd, &read_fds, head); // select incoming connections and handle them as necessary
```

Comment:
let's rename this `select` method to `handle_connection`

Reply:
there doesn't seem to be any advantage of passing read_fds into the `select` method, since it gets recreated from the list on every iteration anyways. we should always try to reduce the scope of our variables as much as possible. instead of defining read_fds out here and passing it as a reference into select, let's just define read_fds inside of the `handle_connection` method